### PR TITLE
AgentMessageType.action[s]

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

This PR includes:
- The change of `AgentMessageType.action` -> `AgentMessageType.actions` (backend code only)
- Adjustment to the front-end to read `AgentMessageType.actions[0]` (actual implementation of multi actions in the front-end will be a different task)
- Adding `Agent.*Action.step` step field to order the actions in a multi actions world
- A migration to update all steps to 0 for  all legacy actions.

## Risk
This is high risk since we are changing the way we read actions. We should thoroughly test it on front-edge before. 

## Deploy Plan
- Block deployments
- Run initdb for front
- Deploy on front-edge
- Test it on front-edge
- Deploy on front
- Unlock deployments.

### CODE REVIEW

File: front/components/assistant/conversation/AgentMessage.tsx [hunk 1]
```diff
@@ -242,7 +242,7 @@ export function AgentMessage({
   }, [
     agentMessageToRender.content,
     agentMessageToRender.status,
-    agentMessageToRender.action,
+    agentMessageToRender.actions,
```
Comment (fontanierh) [inline front/components/assistant/conversation/AgentMessage.tsx:245]: are we sure this is still fine ? Any insight on why this thing is a dependency even though not used above ? Could This be replaced by actions.length or do we mutate the action's content ?

File: front/components/assistant/conversation/AgentMessage.tsx [hunk 2]
```diff
@@ -312,18 +312,18 @@ export function AgentMessage({
   >(null);
   useEffect(() => {
     if (
-      agentMessageToRender.action &&
-      isRetrievalActionType(agentMessageToRender.action) &&
-      agentMessageToRender.action.documents
+      agentMessageToRender.actions.length > 0 &&
```
Comment (fontanierh) [inline front/components/assistant/conversation/AgentMessage.tsx:315]: can we add a `deprecatedGetSingleActionFromAgentMessage` function (or whatever other name as long as it starts with deprecated) so it is easy to find and clean all of that stuff later 🙏

File: front/lib/api/assistant/generation.ts [hunk 3]
```diff
@@ -112,20 +112,20 @@ export async function renderConversationForModel({
       }
       // There are contexts where we want to exclude the actions from the rendering.
       // Eg: During the conversation title generation step.
-      if (m.action && !excludeActions) {
-        if (isRetrievalActionType(m.action)) {
+      if (m.actions.length > 0 && !excludeActions) {
```
Comment (fontanierh) [inline front/lib/api/assistant/generation.ts:115]: Same here let's use `deprecatedGetSingleAction` 🙏

File: front/lib/models/assistant/actions/dust_app_run.ts [hunk 4]
```diff
@@ -157,6 +158,10 @@ AgentDustAppRunAction.init(
       type: DataTypes.JSONB,
       allowNull: true,
     },
+    step: {
```
Comment (fontanierh) [inline front/lib/models/assistant/actions/dust_app_run.ts:161]: Do we want to just set a default 0 for now and remove later ? easier than migrating no ?

**Review by fontanierh — COMMENTED** [review]
Looks great ! just a couple of comments

**Review by lasryaric — COMMENTED** [review]
Addressed all your comments @fontanierh

**Review by fontanierh — APPROVED** [review]
LGTM

File: front/lib/models/assistant/actions/dust_app_run.ts [hunk 5]
```diff
@@ -157,6 +158,10 @@ AgentDustAppRunAction.init(
...
+      type: DataTypes.INTEGER,
+      allowNull: true,
```
Comment (spolu) [inline front/lib/models/assistant/actions/dust_app_run.ts:163]: @lasryaric Let's make sure to remove this switch this allowNull now that this is shipped :pray:

**Pull request merged**